booster/library/Booster/Pattern/Rewrite.hs

```diff
@@ -480,12 +513,6 @@ applyRule pat@Pattern{ceilConditions} rule =
             _other ->
                 pure ()
 
-        -- if a new constraint is going to be added, the equation cache is invalid
-        unless (null newConstraints) $ do
-            withContextFor Equations . logMessage $
-                ("New path condition ensured, invalidating cache" :: Text)
-
-            lift . RewriteT . lift . modify $ \s -> s{equations = mempty}
```

Comment:
(Why) Is this not true any more?

Reply:
We have a similar piece of code in `applyRule`, just after the call to `checkEnsures`. On master we are invalidating the cache twice: at the end of `checkEnsures` (removed here) and right after the call to `checkEnsures` in `applyRule`.